packages/lib/tag/auth.ts

```diff
@@ -0,0 +1,22 @@
+import { hasUserEnvironmentAccess } from "../environment/auth";
+import { getTag } from "./service";
+import { unstable_cache } from "next/cache";
+
+export const canUserAccessTag = async (userId: string, tagId: string): Promise<boolean> =>
+  await unstable_cache(
+    async () => {
+      if (!userId) return false;
```

Comment:
forgot how we did it in the other PRs, but I think input validation makes sense and is a better approach. Since something is fundamentally going wrong if we get a `undefined` here.